### PR TITLE
refactor(apig/response): refactor custom response resoruce

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
refactor resoruce of API custom response.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. refactor resource functions and parameters.
2. refactor acc test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccResponse_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccResponse_ -timeout 360m -parallel 4
=== RUN   TestAccResponse_basic
=== PAUSE TestAccResponse_basic
=== RUN   TestAccResponse_customRules
=== PAUSE TestAccResponse_customRules
=== CONT  TestAccResponse_basic
=== CONT  TestAccResponse_customRules
--- PASS: TestAccResponse_customRules (500.52s)
--- PASS: TestAccResponse_basic (503.41s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig     503.505s
```

### CODE REVIEW

File: huaweicloud/services/apig/resource_huaweicloud_apig_response.go [hunk 1]
```diff
@@ -1,51 +1,81 @@
...
+	ErrorTypeAuthorizerIdentitiesFailure ErrorType = "AUTHORIZER_IDENTITIES_FAILURE"
+	ErrorTypeBackendUnavailable          ErrorType = "BACKEND_UNAVAILABLE"
+	ErrorTypeBackendTimeout              ErrorType = "BACKEND_TIMEOUT"
+	ErrorTypeThrottled                   ErrorType = "THROTTLED"
+	ErrorTypeUnauthorized                ErrorType = "UNAUTHORIZED"
+	ErrorTypeAccessDenied                ErrorType = "ACCESS_DENIED"
+	ErrorTypeNotFound                    ErrorType = "NOT_FOUND"
+	ErrorTypeRequestParametersFailure    ErrorType = "REQUEST_PARAMETERS_FAILURE"
+	ErrorTypeDefault4XX                  ErrorType = "DEFAULT_4XX"
+	ErrorTypeDefault5XX                  ErrorType = "DEFAULT_5XX"
```
Comment (chengxiangdong) [inline huaweicloud/services/apig/resource_huaweicloud_apig_response.go]: The names of these const variables are clear enough, and using the `ErrorType` type seems to introduce unnecessary type conversions.

File: huaweicloud/services/apig/resource_huaweicloud_apig_response.go [hunk 2]
```diff
@@ -1,51 +1,81 @@
...
 			"name": {
 				Type:     schema.TypeString,
 				Required: true,
-				ValidateFunc: validation.StringMatch(
-					regexp.MustCompile("^[A-Za-z0-9_-]{1,64}$"), "The name consists of 1 to 64 characters. "+
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[\w-]*$`),
 						"Only letters, digits, hyphens(-), and underscores (_) are allowed."),
+					validation.StringLenBetween(1, 64),
+				),
```
Comment (chengxiangdong) [inline huaweicloud/services/apig/resource_huaweicloud_apig_response.go:76]:
```suggestion
				ValidateFunc: validation.All(
					validation.StringMatch(regexp.MustCompile(`^[\w-]{1,64}$`),
						"Only letters, digits, hyphens(-), and underscores (_) are allowed."),
				),
```

It is more efficient to use one validation.

File: docs/resources/apig_response.md [hunk 3]
```diff
@@ -30,24 +30,27 @@ resource "huaweicloud_apig_response" "test" {
...
 
-* `region` - (Optional, String, ForceNew) Specifies the region in which to create the API custom response resource. If
-  omitted, the provider-level region will be used. Changing this will create a new API custom response resource.
+* `region` - (Optional, String, ForceNew) Specifies the region where the API custom response is located.  
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
 
-* `group_id` - (Required, String, ForceNew) Specifies the ID of the API group to which the API response belongs to.
-  Changing this will create a new API custom response resource.
+* `instance_id` - (Required, String, ForceNew) Specifies the ID of the dedicated instance to which the API group and the
+  API custom response belongs.
```
Comment (Jason-Zhang9309) [inline docs/resources/apig_response.md]: s/belongs/belong

File: docs/resources/apig_response.md [hunk 4]
```diff
@@ -30,24 +30,27 @@ resource "huaweicloud_apig_response" "test" {
...
 
-* `instance_id` - (Required, String, ForceNew) Specifies the ID of the APIG dedicated instance to which the API group
-  where the API custom response belongs. Changing this will create a new API custom response resource.
+* `group_id` - (Required, String, ForceNew) Specifies the ID of the API group to which the API custom response
+  belongs.  
+  Changing this will create a new resource.
 
-* `name` - (Required, String) Specifies the name of the API custom response. The name consists of 1 to 64 characters,
-  and only letters, digits, hyphens(-), and underscores (_) are allowed.
+* `name` - (Required, String) Specifies the name of the API custom response.
```
Comment (Jason-Zhang9309) [inline docs/resources/apig_response.md:44]: extral spcace at the end of  this line

File: docs/resources/apig_response.md [hunk 5]
```diff
@@ -30,24 +30,27 @@ resource "huaweicloud_apig_response" "test" {
...
+  Changing this will create a new resource.
 
-* `name` - (Required, String) Specifies the name of the API custom response. The name consists of 1 to 64 characters,
-  and only letters, digits, hyphens(-), and underscores (_) are allowed.
+* `name` - (Required, String) Specifies the name of the API custom response.  
+  The valid length is limited from `1` to `64`, letters, digits and hyphens (-) and underscores (_) are allowed.
 
-* `rule` - (Optional, List) Specifies the API custom response rules definition. The object structure is documented
-  below.
+* `rule` - (Optional, List) Specifies the API custom response rules definition.
```
Comment (Jason-Zhang9309) [inline docs/resources/apig_response.md:47]: extral spcace at the end of  this line

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_response_test.go [hunk 6]
```diff
@@ -197,15 +193,15 @@ resource "huaweicloud_apig_response" "test" {
     status_code = 401
   }
 }
-`, testAccApigResponse_base(rName), rName)
+`, testAccResponse_base(name), name)
 }
 
-func testAccApigResponse_update(rName string) string {
+func testAccResponse_update(name string) string {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_response_test.go]: the names of base resources should not be updated

Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_response_test.go]: There is no test function to reference this function.

File: docs/resources/apig_response.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/apig_response.md]: fixed.

File: huaweicloud/services/apig/resource_huaweicloud_apig_response.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_response.go:76]: The current writing method is better to prompt when customer make a mistake on format.

File: huaweicloud/services/apig/resource_huaweicloud_apig_response.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_response.go]: sure.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_response_test.go (hunk 6 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_response_test.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccResponse_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccResponse_ -timeout 360m -parallel 4
=== RUN   TestAccResponse_basic
=== PAUSE TestAccResponse_basic
=== RUN   TestAccResponse_customRules
=== PAUSE TestAccResponse_customRules
=== CONT  TestAccResponse_basic
=== CONT  TestAccResponse_customRules
--- PASS: TestAccResponse_basic (527.79s)
--- PASS: TestAccResponse_customRules (558.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      559.051s
```

File: docs/resources/apig_response.md [hunk 7]
```diff
@@ -73,13 +76,13 @@ The `rule` block supports:
...
+* `updated_at` - The latest update time of the API custom response.
 
 ## Import
 
 API Responses can be imported using their `name` and IDs of the APIG dedicated instances and API groups to which the API
-response belongs, separated by a slash, e.g.
+response belongs, separated by slashes, e.g.
 
 ```
 $ terraform import huaweicloud_apig_response.test <instance id>/<group id>/<name>
```
Comment (ShiChangkuo) [inline docs/resources/apig_response.md]:
```suggestion
```bash
terraform import huaweicloud_apig_response.test <instance id>/<group id>/<name>
```

Comment (Lance52259) [reply docs/resources/apig_response.md]: done.

**Pull request merged**